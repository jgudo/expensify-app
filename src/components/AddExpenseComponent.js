import React from 'react';
import {connect} from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { addExpense } from '../actions/expenses';

export class AddExpenseComponent extends React.Component {
    onSubmit = (expense) => {
        this.props.onSubmit(expense);
        //props.dispatch(addExpense(expense));
        this.props.history.push('/');
    };

    render() {
        return (
            <div>
                 <h1>Add New Expense Form</h1>
                 <ExpenseForm 
                     onSubmit={this.onSubmit}
                 />
            </div>
         
         );
    }
}


const mapDispatchToProps = (dispatch) => ({
    onSubmit: (expense) => dispatch(addExpense(expense))
});

export default connect(undefined, mapDispatchToProps)(AddExpenseComponent);