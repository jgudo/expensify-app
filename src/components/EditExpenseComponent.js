import React from 'react';
import {connect} from 'react-redux';
import  ExpenseForm from './ExpenseForm';
import {startEditExpense, startRemoveExpense} from '../actions/expenses';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

export class EditExpenseComponent extends React.Component {

    onSubmit = (expense) => {
        this.props.startEditExpense(this.props.expense.id, expense);
        this.props.history.push('/');
        
    }
    // onRemove = () => {
    //     this.props.removeExpense({id: this.props.expense.id});
    //     this.props.history.push('/');
    //    
    // }

    render() {
        return(
            <div className="page-header">
                <div className="content-container">
                    <div className="page-header__action">
                        <h1 className="page-header__title">Edit Expense</h1>
                        <button 
                            className="button button--secondary button--action"
                            onClick={() => {
                            this.props.startRemoveExpense({id: this.props.expense.id});
                            this.props.history.push('/');
                            //props.dispatch(removeExpense({id:props.expense.id}));
                        }}
                        >
                            <span>Remove</span>
                            <span><FontAwesomeIcon icon={['fas', 'trash']} /> </span>
                    
                        </button>
                    </div>
                </div>
                <div className="content-container">
                    <ExpenseForm
                        expense={this.props.expense}
                        onSubmit={this.onSubmit}
                    />
                </div> 
            </div>
            
        );
    }
}
const mapDispatchToProps = (dispatch) => ({
    startEditExpense: (id,expense) => dispatch(startEditExpense(id, expense)),
    startRemoveExpense: (data) => dispatch(startRemoveExpense(data))
});

const mapStateToProps = (state, props) => {
    return {
        expense: state.expenses.find( (expense) => expense.id === props.match.params.id)
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(EditExpenseComponent);
