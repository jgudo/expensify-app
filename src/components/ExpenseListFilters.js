import React from 'react';
import { connect } from 'react-redux';
import {DateRangePicker} from 'react-dates';
import {setTextFilter, sortByDate, sortByAmount, setStartDate, setEndDate} from '../actions/filters';

export class ExpenseListFilters extends React.Component {
    state = {
        calendarFocused: null
    }

    onDatesChange = ({startDate, endDate}) => {
        this.props.setStartDate(startDate);
        this.props.setEndDate(endDate);
        //console.log('STARTDATE:', startDate, ' END DATE: ', endDate);
    }

    onFocusChange = (calendarFocused) => {
        this.setState(() => ({calendarFocused }));
    }

    onTextChange = (e) => {
        this.props.setTextFilter(e.target.value)
    }

    onSelectChange = (e) => {
        if(e.target.value === 'date') {
            this.props.sortByDate();
        } else if(e.target.value === 'amount') {
            this.props.sortByAmount();
        }
    }

    render() {
        return (
            <div className="content-container">
                <div className="input-group">
                    <div className="input-group__item">
                        <input type="text" 
                            className="text-input"
                            placeholder="Search Filter"
                            value={this.props.filters.text} 
                            onChange={this.onTextChange}
                        />
                    </div>
                    <div className="input-group__item">
                        <select onChange={this.onSelectChange} className="select">
                            <option value="date">Date</option>
                            <option value="amount">Amount</option>
                        </select>
                    </div>
                    <div className="input-group__item">
                        <DateRangePicker 
                            startDate={this.props.filters.startDate}
                            startDateId="start_date"
                            endDate={this.props.filters.endDate}
                            endDateId="end_date"
                            onDatesChange={this.onDatesChange}
                            focusedInput={this.state.calendarFocused}
                            onFocusChange={this.onFocusChange}
                            showClearDates={true}
                            numberOfMonths={1}
                            isOutsideRange={ () => false}
                        />
                    </div>
                </div>
                
                
                
            </div>
        );
    }
}

const mapDispatchToProps = (dispatch) => ({
    setTextFilter: (text) => dispatch(setTextFilter(text)),
    sortByDate: () => dispatch(sortByDate()),
    sortByAmount: () => dispatch(sortByAmount()),
    setStartDate: (startDate) => dispatch(setStartDate(startDate)),
    setEndDate: (endDate) => dispatch(setEndDate(endDate))  
});

const mapStateToProps = (state) => {
    return {
        filters: state.filters
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(ExpenseListFilters);