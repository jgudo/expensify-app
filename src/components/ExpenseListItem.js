import React from 'react';
import {Link} from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral';

const ExpenseListItem = ({id,description, amount, createdAt}) =>  (
    <Link to={`/edit/${id}`} className="list-item">
        <div className="list-item__description">
            <h3 className="list-item__title">{description}</h3>
            <span>{moment(createdAt).format('MMMM Do YYYY')}</span>
        </div>
        <div>
            <h3> ₱{numeral(amount / 100).format('₱0,0.00')}</h3>
        </div>
    </Link>
);

export default ExpenseListItem;
//export default ExpenseListItem;