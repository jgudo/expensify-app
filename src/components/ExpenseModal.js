import React from 'react';
import Modal from 'react-modal';

const ExpenseModal = (props) => {
    return (
        <Modal
            isOpen={props.modalIsOpen}
            onRequestClose={props.closeModal}
            closeTimeoutMS={200}
            className="modal"
            contentLabel="Remove Expense Modal"
        >
            <div>
                <h2 className="modal__title">Remove this expense?</h2>
                <button onClick={props.onRemove} className="modal__button">Yes</button>
                <button onClick={props.closeModal} className="modal__button">No</button>
            </div>
            
        </Modal>
    );
}
export default ExpenseModal;