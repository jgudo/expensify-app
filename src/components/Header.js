import React from 'react';
import {connect} from 'react-redux';
import { startLogout} from '../actions/auth';
import {Link} from 'react-router-dom';
import Modal from 'react-modal';

export class Header extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            modalIsOpen: false
        };
    }

    openModal = () => {
        this.setState( () => ({
            modalIsOpen: true
        }));
    };
    

    closeModal = () => {
        this.setState( () => ({
            modalIsOpen: false
        }));
    };

    render() {
        return (
            <header className="header">
                <div className="content-container">
                    <div className="header__content">
                        <Link to="/dashboard" className="header__title">
                            <h1>Expensify</h1>
                        </Link>
                        <button 
                            onClick={this.openModal} 
                            className="button button--transparent-l">
                            <span>Logout</span>
                        </button>
                    </div>
                </div>
                <Modal
                    isOpen={this.state.modalIsOpen}
                    onAfterOpen={this.afterOpenModal}
                    onRequestClose={this.closeModal}
                    closeTimeoutMS={200}
                    className="modal"
                    contentLabel="Logout modal"
                >
                    <div>
                        <h2 className="modal__title">Sure to logout?</h2>
                        <button onClick={this.props.startLogout} className="modal__button">Yes</button>
                        <button onClick={this.closeModal} className="modal__button">No</button>
                    </div>
                    
                </Modal>
            </header>
        );
    }
}

const mapDispatchToProps = (dispatch) => ({
    startLogout: () => dispatch(startLogout())
});

export default connect(undefined, mapDispatchToProps)(Header);