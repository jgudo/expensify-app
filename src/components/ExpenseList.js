import React from 'react';
import {connect} from 'react-redux';
import ExpenseListItem from './ExpenseListItem';
import selectExpenses from '../selectors/expenses';

export const ExpenseList = (props) => (
    <div>
        {props.expenses.length === 0 ?  (
            <p>No Expenses List</p>
        ) : (
            props.expenses.map( (item) => {
                return <ExpenseListItem key={item.id} {...item}/>
            })
        )}
     
    </div>
);

const mapStateToProps = (state) => {
    return {
        expenses: selectExpenses(state.expenses, state.filters)
    };
};
export default connect(mapStateToProps)(ExpenseList);

