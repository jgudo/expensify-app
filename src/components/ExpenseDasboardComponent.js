import React from 'react';
import ExpenseList from './ExpenseList';
import ExpenseListFilters from './ExpenseListFilters';

const ExpenseDashboardComponent = () => (
    <div>
        <ExpenseListFilters />
        <ExpenseList />
    </div>
);

export default ExpenseDashboardComponent;