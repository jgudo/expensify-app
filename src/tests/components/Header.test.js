import React from 'react';
import {shallow} from 'enzyme';
//import ReactShallowRenderer from 'react-test-renderer/shallow';
import {Header} from '../../components/Header';

test('Should render correctly', () => {
    const wrapper = shallow(<Header startLogout={ () => {} }/>);
    expect(wrapper).toMatchSnapshot();
   
    // const renderer = new ReactShallowRenderer();
    // renderer.render(<Header />);
    // expect(renderer.getRenderOutput()).toMatchSnapshot();
});

test('Should call startLogout', () => {
    const startLogout = jest.fn();
    const wrapper = shallow(<Header startLogout={startLogout}/>);
    wrapper.find('button').simulate('click');
   
    expect(startLogout).toHaveBeenCalled();

});


