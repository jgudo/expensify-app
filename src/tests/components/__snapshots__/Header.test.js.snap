// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render correctly 1`] = `
<header>
  <h1>
    Expensify
  </h1>
  <ul>
    <li>
      <NavLink
        activeClassName="is-active"
        ariaCurrent="true"
        exact={true}
        to="/"
      >
        Home
      </NavLink>
    </li>
    <li>
      <NavLink
        activeClassName="is-active"
        ariaCurrent="true"
        to="/create"
      >
        Create Expense
      </NavLink>
    </li>
    <li>
      <NavLink
        activeClassName="is-active"
        ariaCurrent="true"
        to="/help"
      >
        Get Help
      </NavLink>
    </li>
    <li>
      <button
        onClick={[Function]}
      >
        Logout
      </button>
    </li>
  </ul>
</header>
`;
